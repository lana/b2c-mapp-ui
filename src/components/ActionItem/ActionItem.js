import TextParagraph from '../TextParagraph/TextParagraph.vue';

const errorStatuses = [
  'rejected',
  'expired',
];

const components = {
  TextParagraph,
};

const props = {
  title: {
    type: String,
    default: '',
  },
  description: {
    type: String,
    default: '',
  },
  status: {
    type: String,
    default: '',
  },
  highlight: Boolean,
  dataTestId: {
    type: String,
    default: 'action-item',
  },
};

const emits = ['click'];

const computed = {
  hasErrorStatus() {
    if (!this.status) { return; }
    const result = errorStatuses.includes(this.status.toLowerCase());
    return result;
  },
  descriptionTextColor() {
    if (!this.hasErrorStatus) { return; }
    return 'red-500';
  },
};

const methods = {
  onClick(event) {
    this.$emit('click', event);
  },
};

const name = 'ActionItem';

const ActionItem = {
  name,
  components,
  computed,
  props,
  emits,
  methods,
};

export default ActionItem;
