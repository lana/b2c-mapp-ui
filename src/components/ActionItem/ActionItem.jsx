import PropTypes from 'prop-types';

import CSS from './styles.css';
import TextParagraph from '../TextParagraph/TextParagraph';

const ActionItem = ({ onClick, mediaContent, mediaColor, highlight, className, title, dataTestId }) => {
  const mediaColorClass = (mediaColor) ? CSS[mediaColor] : '';
  const highLighClass = (highlight) ? `${CSS.highlight}` : '';

  return (
    <li data-testid={dataTestId} onClick={onClick ? (event) => onClick(event) : false} className={`${CSS.item} ${className}`}>
      {mediaContent && <div data-testid={`${dataTestId}-mediacolor`} className={`${CSS.media} ${mediaColorClass}`}>{mediaContent}</div>}
      <div className={CSS.body}>
        <TextParagraph dataTestId={`${dataTestId}-highlight`} className={highLighClass}>{title}</TextParagraph>
      </div>
    </li>
  );
};

ActionItem.defaultProps = {
  dataTestId: 'action-item',
  className: '',
  onClick: null,
  mediaContent: null,
  mediaColor: false,
  highlight: false,
};

ActionItem.propTypes = {
  onClick: PropTypes.func,
  mediaContent: PropTypes.node,
  mediaColor: PropTypes.bool,
  highlight: PropTypes.bool,
  dataTestId: PropTypes.string,
  className: PropTypes.string,
  title: PropTypes.string.isRequired,
};

export default ActionItem;
