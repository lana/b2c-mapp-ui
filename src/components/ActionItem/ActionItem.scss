@import '../../styles/colors';

$item-box-size: 48px;

.item {
  display: flex;
  flex-direction: row;
  justify-content: flex-start;
  align-items: center;
  overflow: hidden;
  position: relative;
  height: $item-box-size;
  padding: 8px 16px;
}

.media {
  display: flex;
  justify-content: center;
  align-items: center;
  flex-shrink: 0;
  overflow: hidden;
  margin-right: 16px;
  width: $item-box-size;
  height: 100%;
  border-radius: 14px;
  background-color: $black-100;
  border: solid 1px $black-300;

  .icon {
    width: 40px;
    height: 40px;
  }
}

.body {
  height: 100%;
  display: flex;
  flex-direction: column;
  align-items: flex-start;
  justify-content: center;
  flex-grow: 1;
}

.icon {
  flex-shrink: 0;
}

.title {
  font-size: 16px;
  line-height: 24px;
  color: $black-700;

  &.highlight {
    color: $blue-500;
  }
}
