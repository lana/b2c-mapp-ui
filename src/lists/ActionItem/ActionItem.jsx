import PropTypes from 'prop-types';

import CSS from './styles.css';
import Text from '../../typography/Text/Text';

const ActionItem = ({ onClick, mediaContent, mediaColor, highlight, className, title, dataTestId }) => {
	const mediaColorClass = mediaColor ? CSS[mediaColor] : '';
	const highLighClass = highlight ? `${CSS['highlight']}` : '';

	return (
		<li data-testid={dataTestId} onClick={onClick ? e => onClick(e) : false} className={`${CSS.item} ${className}`}>
			{mediaContent && (
				<div data-testid={`${dataTestId}-mediacolor`} className={`${CSS.media} ${mediaColorClass}`}>
					{mediaContent}
				</div>
			)}
			<div className={CSS.body}>
				<Text dataTestId={`${dataTestId}-highlight`} className={highLighClass}>
					{title}
				</Text>
			</div>
		</li>
	);
};

ActionItem.defaultProps = {
	dataTestId: 'action-item',
	className: '',
	onClick: null,
	mediaContent: null,
	mediaColor: false,
	highlight: false,
};

ActionItem.propTypes = {
	onClick: PropTypes.func,
	mediaContent: PropTypes.node,
	mediaColor: PropTypes.bool,
	highlight: PropTypes.bool,
	dataTestId: PropTypes.string,
	className: PropTypes.string,
	title: PropTypes.string.isRequired,
};

export default ActionItem;
