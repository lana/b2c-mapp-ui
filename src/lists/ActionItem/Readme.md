# ActionItem
A list item which takes the consumer to perform an action in another screen.

| Prop |  Value Type | Required |
| --- | --- | --- |
| `className` | `String` | false | 
| `media` | `ReactNode` | false | 
| `mediaColor` | `blue`, `green`, `yellow`, `brown`, `purple` | false | 
| `title` | `String` | **true** | 
| `onClick` | `(e: HTMLMouseEvent) => void` | false | 
| `highlight` | `Boolean` | false | 

```
<ul>
    <ActionItem title='1st item' media={<img src="https://source.unsplash.com/random/44x44" />} />
    <ActionItem title='2nd item' media={<ExampleIcon/>} highlight mediaColor="blue" />
    <ActionItem title='3nd item' media={<ExampleIcon />} highlight mediaColor="purple" />
    <ActionItem title='4rd item' media={<ExampleIcon/>} />
</ul>
```




