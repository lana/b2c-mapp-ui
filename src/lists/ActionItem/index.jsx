import CSS from './styles.css'
import Text from '../../typography/Text'

// eslint-disable-next-line react/display-name
export default ({ onClick, media, mediaColor, highlight, className, title }) => {
	const mediaColorClass = mediaColor ? CSS[mediaColor] : ''
	const highLighClass = highlight ? `${CSS['highlight']}` : ''
	return (
		<li onClick={onClick ? e => onClick(e) : false} className={`${CSS.item} ${className || ''}`}>
			{media && <div className={`${CSS.media} ${mediaColorClass}`}>{media}</div>}
			<div className={CSS.body}>
				<Text className={highLighClass}>{title}</Text>
			</div>
		</li>
	)
}
