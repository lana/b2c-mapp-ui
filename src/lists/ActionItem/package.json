{
  "name": "ActionItem",
  "version": "0.0.0",
  "private": true,
  "main": "./ActionItem.jsx"
}
