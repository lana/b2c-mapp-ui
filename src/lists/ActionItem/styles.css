.item {
	display: flex;
	flex-direction: row;
	justify-content: flex-start;
	align-items: center;
	overflow: hidden;
	position: relative;
	height: 44px;
	padding: 8px 16px;
}

.media {
	display: flex;
	justify-content: center;
	align-items: center;
	flex-shrink: 0;
	overflow: hidden;
	margin-right: 12px;
	width: 44px;
	height: 100%;
	border-radius: 3px;
	background: var(--black-400);
}

.media.blue { background-color: var(--blue-300); }
.media.yellow { background-color: var(--yellow-300); }
.media.green { background-color: var(--green-300); }
.media.brown { background-color: var(--brown-300); }
.media.purple { background-color: var(--purple-300); }

.highlight {
	color: var(--blue-500)
}
.body {
	height: 100%;
	display: flex;
	flex-direction: column;
	align-items: start;
	justify-content: center;
	flex-grow: 1;
}

.icon {
	flex-shrink: 0;
}
