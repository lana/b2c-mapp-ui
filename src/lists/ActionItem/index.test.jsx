import { mount } from 'enzyme';
import ActionItem from './index'
import { shallow } from 'preact-render-spy';

describe('UI/lists/ActionItem', () => {
	let wrapper;

	test('should render correctly',  () => {
		wrapper = mount(<ActionItem title='boo' media={<img src="any" />}  />)
		expect(wrapper).toMatchSnapshot()
	})
	test('should render correctly with params highlight and mediaColor',  () => {
		wrapper = mount(<ActionItem title='boo' media={<img src="any" highlight mediaColor="blue" />}  />)
		expect(wrapper).toMatchSnapshot()
	})
	test('should find declared params: media, mediaColor, highlight, className, title',  () => {
		const div = document.createElement('div');
		const props = {
			className: "CLASSNAME",
			mediaColor: "MEDIACOLOR",
			media: "MEDIA",
			highlight: true,
			title: "TITLE"
		}
		wrapper = mount(<ActionItem {...props} />)
		// workaround to avoid preact/enzyme conflict
		div.innerHTML = wrapper.html()
		expect(div.querySelectorAll(`li.${props.className}`).length).toBe(1)
		expect(div.querySelectorAll(`div.${props.mediaColor}`).length).toBe(1)
		expect(div.querySelector(`div.${props.mediaColor}`).innerHTML).toBe(props.media)
		expect(div.querySelector(`p.highlight`).innerHTML).toBe(props.title)
	})
	test('should trigger component\'s onClick', () => {
		let check = false;
		wrapper = shallow(<ActionItem onClick={() => { check = true; }} />)
		wrapper.find('[onClick]').simulate('click')
		expect(check).toBe(true)
	})

});
