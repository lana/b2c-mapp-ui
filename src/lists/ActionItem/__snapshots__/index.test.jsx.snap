// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UI/lists/ActionItem should render correctly 1`] = `ReactWrapper {}`;

exports[`UI/lists/ActionItem should render correctly with params highlight and mediaColor 1`] = `ReactWrapper {}`;
